flang/lib/Optimizer/Transforms/AddDebugInfo.cpp

```diff
@@ -52,21 +55,40 @@ void AddDebugInfoPass::runOnOperation() {
   mlir::ModuleOp module = getOperation();
   mlir::MLIRContext *context = &getContext();
   mlir::OpBuilder builder(context);
-  std::string inputFilePath("-");
-  if (auto fileLoc = module.getLoc().dyn_cast<mlir::FileLineColLoc>())
-    inputFilePath = fileLoc.getFilename().getValue();
+  llvm::StringRef fileName;
+  std::string filePath;
```

Comment:
Does DWARF support paths with wide characters?

Reply:
`DWARF` has `DW_AT_use_UTF8` attribute which can be used to say that program entities and filenames use `UTF-8` representation.

Reply:
I see that some of the functions in `llvm::sys::fs` represent paths as `Twine`s so I guess a `std::string` should be good enough for now. Off the top of my head, I can't think of a reason why a `std::string` can't store a UTF8 string (::size() might be confusing, but that shouldn't matter here).